File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"
#include <stdlib.h>

char *to_rna(const char *dna){
    int i = 0;
    char *rna = calloc(100, sizeof(char));
    while (dna[i]){
        switch (dna[i]){
                case 'G':
                    rna[i] = 'C';
                    break;
                case 'C':
                    rna[i] = 'G';
                    break;
                case 'T':
                    rna[i] = 'A';
                    break;
                case 'A':
                    rna[i] = 'U';
                    break;
        }
        i++;
    }
    return rna;
}

File: solutions/julia/rna-transcription/1/rna-transcription.jl
const NUCLEOTIDES = ('G', 'C', 'T', 'A')

const TRANSCRIPTION = Dict(
    'G' => 'C',
    'C' => 'G',
    'T' => 'A',
    'A' => 'U'
)

function to_rna(dna)
    rna = []
    for tide in dna
        if tide ∉ NUCLEOTIDES; throw(ErrorException("")); end
        push!(rna, TRANSCRIPTION[tide])
    end
    join(rna)
end